### PR TITLE
test timeout_to_kill

### PR DESCRIPTION
**Pull request by pondrejk**
### Problem Statement
timeout to kill setting should cancel the rex job if it takes too long

### Solution
adding test

### CODE REVIEW

**Comment by pondrejk** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/cli/test_remoteexecution.py -k timeout
theforeman:
    smart_proxy_remote_execution_ssh: 115
```

**Review by vsedmik — APPROVED** [review]
Codewisely looks good, packit/PRT pending.

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7096
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_remoteexecution.py -k timeout --external-logging
Test Result : ========== 1 failed, 59 deselected, 72 warnings in 2178.44s (0:36:18) ==========
```

File: tests/foreman/cli/test_remoteexecution.py [hunk 1]
```diff
@@ -134,6 +134,40 @@ def test_positive_run_default_job_template(
...
+        :parametrized: yes
+        """
+        client = rex_contenthost
+        command = 'sleep 160'
+        invocation_command = module_target_sat.cli_factory.job_invocation(
+            {
+                'job-template': 'Run Command - Script Default',
+                'inputs': f'command={command}',
+                'search-query': f"name ~ {client.hostname}",
+                'timeout_to_kill': '5',
```
Comment (vsedmik) [inline tests/foreman/cli/test_remoteexecution.py:155]: Per PRT failure I guess this could help
```suggestion
                'timeout-to-kill': '5',
```

**Review by vsedmik — CHANGES_REQUESTED** [review]

**Comment by pondrejk** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/cli/test_remoteexecution.py -k timeout
theforeman:
    smart_proxy_remote_execution_ssh: 115
```

**Comment by pondrejk** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/cli/test_remoteexecution.py -k timeout
theforeman:
    smart_proxy_remote_execution_ssh: 115
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7131
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_remoteexecution.py -k timeout --external-logging
Test Result : ========== 59 deselected, 67 warnings, 1 error in 1719.57s (0:28:39) ===========
```

**Comment by pondrejk** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/cli/test_remoteexecution.py -k timeout
theforeman:
    smart_proxy_remote_execution_ssh: 115
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7156
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_remoteexecution.py -k timeout --external-logging
Test Result : ========== 1 failed, 59 deselected, 72 warnings in 2155.09s (0:35:55) ==========
```

**Comment by pondrejk** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/cli/test_remoteexecution.py -k timeout
theforeman:
    smart_proxy_remote_execution_ssh: 115
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7196
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_remoteexecution.py -k timeout --external-logging
Test Result : ========== 1 failed, 59 deselected, 72 warnings in 2225.42s (0:37:05) ==========
```

**Comment by pondrejk** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/cli/test_remoteexecution.py -k timeout
theforeman:
    smart_proxy_remote_execution_ssh: 115
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7245
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_remoteexecution.py -k timeout --external-logging
Test Result : ========== 1 failed, 59 deselected, 72 warnings in 2239.36s (0:37:19) ==========
```

**Review by lhellebr — APPROVED** [review]

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_remoteexecution.py -k timeout

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_remoteexecution.py -k timeout

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7356
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_remoteexecution.py -k timeout --external-logging
Test Result : =========== 59 deselected, 72 warnings, 1 error in 861.66s (0:14:21) ===========
```

**Comment by pondrejk** [comment]
eh, prt won't pass due to the current registration issue, must await sn 62

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_remoteexecution.py -k timeout

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7401
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_remoteexecution.py -k timeout --external-logging
Test Result : ========== 1 failed, 59 deselected, 72 warnings in 1040.12s (0:17:20) ==========
```

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_remoteexecution.py -k timeout

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7636
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_remoteexecution.py -k timeout --external-logging
Test Result : ========== 1 failed, 59 deselected, 72 warnings in 1273.61s (0:21:13) ==========
```

**Comment by pondrejk** [comment]
hm, awaiting rubygem-smart_proxy_remote_execution_ssh-0.10.6 to land at stream, having the proper fix to the original issue, stay tuned

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_remoteexecution.py -k timeout

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7804
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_remoteexecution.py -k timeout --external-logging
Test Result : ========== 1 failed, 59 deselected, 72 warnings in 1095.80s (0:18:15) ==========
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7808
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_remoteexecution.py -k timeout --external-logging
Test Result : ========== 1 failed, 59 deselected, 72 warnings in 942.15s (0:15:42) ===========
```

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_remoteexecution.py -k timeout

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7832
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_remoteexecution.py -k timeout --external-logging
Test Result : ========== 1 failed, 59 deselected, 72 warnings in 928.00s (0:15:27) ===========
```

File: tests/foreman/cli/test_remoteexecution.py [hunk 2]
```diff
@@ -134,6 +134,42 @@ def test_positive_run_default_job_template(
...
+        :id: 580f886b-ac24-4afa-9aca-e9afc5cfdc9c
+
+        :expectedresults: Verify the job was killed after specified times
+
+        :parametrized: yes
+
+        :Verifies: SAT-25243
+        """
+        client = rex_contenthost
+        command = r'for i in \`seq 160\`; do sleep \$i; done'
```
Comment (pnovotny) [inline tests/foreman/cli/test_remoteexecution.py:151]: (non-blocking for the test functionality)
With `sleep $i`, the sleeps will be incremental, i.e. sleep 1; sleep 2; sleep 3, etc. 
For sleeping for 160 seconds in 1-second ticks, running `for run in {1..160}; do sleep 1; done` will do the trick.

**Review by pnovotny — APPROVED** [review]
LGTM. My comment above is non-blocking.

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_remoteexecution.py -k timeout

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7936
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/cli/test_remoteexecution.py -k timeout --external-logging
Test Result : ========== 1 passed, 59 deselected, 72 warnings in 964.38s (0:16:04) ===========
```

**Pull request merged**